src/libraries/System.Memory/tests/Span/EnumerateRunePositions.cs

```diff
@@ -0,0 +1,206 @@
+                Assert.False(((object)runePosition).Equals(expected));
+
+                Assert.False(expected == runePosition);
+                Assert.False(runePosition == expected);
+
+                Assert.True(expected != runePosition);
+                Assert.True(runePosition != expected);
+            }
+        }
+
+        private static void RunePosition_TestDeconstruct(RunePosition runePosition)
+        {
+            {
+                (Rune rune, int startIndex) = runePosition;
+                Assert.Equal(runePosition.Rune, rune);
+                Assert.Equal(runePosition.StartIndex, startIndex);
+            }
+            {
+                (Rune rune, int startIndex, int length) = runePosition;
+                Assert.Equal(runePosition.Rune, rune);
+                Assert.Equal(runePosition.StartIndex, startIndex);
+                Assert.Equal(runePosition.Length, length);
+            }
+        }
+
+        [Fact]
+        public static void RunePosition_DefaultTest()
+        {
+            RunePosition runePosition = default;
+            RunePosition_TestProps(default, 0, 0, false, runePosition);
+            RunePosition_TestEquals(default, runePosition);
+            RunePosition_TestDeconstruct(runePosition);
+
+            runePosition = new RunePosition();
+            RunePosition_TestProps(default, 0, 0, false, runePosition);
+            RunePosition_TestEquals(default, runePosition);
+            RunePosition_TestDeconstruct(runePosition);
+        }
+
+        [Fact]
+        public static void EnumerateRunePositions_Empty()
+        {
+            {
+                RunePosition.Utf16Enumerator enumerator = RunePosition.EnumerateUtf16([]).GetEnumerator();
+                Assert.False(enumerator.MoveNext());
+            }
+            {
+                RunePosition.Utf8Enumerator enumerator = RunePosition.EnumerateUtf8([]).GetEnumerator();
+                Assert.False(enumerator.MoveNext());
+            }
+        }
+
+        [Theory]
+        [InlineData(new char[0])] // empty
+        [InlineData(new char[] { 'x', 'y', 'z' })]
+        [InlineData(new char[] { 'x', '\uD86D', '\uDF54', 'y' })] // valid surrogate pair
+        [InlineData(new char[] { 'x', '\uD86D', 'y' })] // standalone high surrogate
+        [InlineData(new char[] { 'x', '\uDF54', 'y' })] // standalone low surrogate
+        [InlineData(new char[] { 'x', '\uD86D' })] // standalone high surrogate at end of string
+        [InlineData(new char[] { 'x', '\uDF54' })] // standalone low surrogate at end of string
+        [InlineData(new char[] { 'x', '\uD86D', '\uD86D', 'y' })] // two high surrogates should be two replacement chars
+        [InlineData(new char[] { 'x', '\uFFFD', 'y' })] // literal U+FFFD
+        public static void EnumerateRunePositions_Battery16(char[] chars)
+        {
+            // Test data is smuggled as char[] instead of straight-up string since the test framework
```

Comment:
Would it make sense to make this test method take a list of RunePositions to check against instead of manually decoding?